reference/security.rst

```diff
@@ -0,0 +1,252 @@
+- **Key Type/Size**: RSA 4096 bits
+- **Signing Algorithm**: RS256
+
+The following Go packages are used:
+
+- ``github.com/lestrrat-go/jwx/v2/jwa``
+- ``github.com/lestrrat-go/jwx/v2/jwt``
+- ``github.com/lestrrat-go/jwx/v2/jwk``
+
+User Sessions
+-------------
+
+CLI-Based Sessions
+^^^^^^^^^^^^^^^^^^
+
+When authenticating the Juju CLI to JAAS, users go through an OAuth login flow, 
+after which the CLI is issued a JWT session token. This token is stored on the 
+filesystem and sent with each request, eliminating the need for repeated logins.
+
+While Juju controller tokens use asymmetric cryptography, session tokens for 
+users are signed using symmetric cryptography (i.e., a shared secret) with a 
+cryptographic hash function.
+
+The shared secret is generated by the JIMM charm operator using Juju secrets
+and passed to the application.
+
+JWTs for CLI sessions contain the user's email address, so they must be secured 
+to prevent information leakage and malicious account use.
+
+Details:
+
+- **Key Size**: >=512 bits
+- **Signing Algorithm**: HS256
+
+The following Go packages are used:
+
+- ``github.com/lestrrat-go/jwx/v2/jwt``
+- ``github.com/lestrrat-go/jwx/v2/jwa``
+
+Browser Cookies
+^^^^^^^^^^^^^^^
+
+In the Juju dashboard, session management is handled using cookies, which store 
+cryptographically encoded values that allow the server to retrieve session data.
+Like CLI sessions, JAAS uses symmetric cryptography with a shared secret.
+
+Unlike the CLI session tokens, browser session cookies only store an encoded 
+session ID rather than personal user information like email addresses.
+While they do not store user information, they must still be kept safe to prevent 
+malicious account use.
+
+The shared secret is generated by the JIMM charm operator using Juju secrets
+and passed to the application.
+
+Details:
+
+- **Key Size**: >=512 bits
+- **Signing Algorithm**: HMAC-SHA256
+
+The following Go packages are used:
+
+- ``github.com/gorilla/sessions``
+- ``github.com/antonlindstrom/pgstore``
+
+OIDC Authentication
+-------------------
+
+JAAS uses OAuth 2.0 and OpenID Connect (OIDC) for user authentication. 
+You can learn more from this `overview <https://developer.okta.com/docs/concepts/oauth-openid/>`__.
+
+The following Go packages are used:
+
+- ``golang.org/x/oauth2``
+- ``golang.org/x/oauth2/clientcredentials``
+- ``github.com/coreos/go-oidc/v3/oidc``
+
+Authorisation Code Flow
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In a browser-based login, users follow the `authorisation code flow <https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow>`__.
+Upon successful login, the access and refresh tokens are stored on the backend, 
+and a session cookie is issued to the user's browser.
+
+To protect against `CSRF attacks <https://auth0.com/docs/secure/attack-protection/state-parameters>`__,
+the backend issues a random nonce in the ``state`` parameter of the OAuth flow.
```

Comment:
utilises the state parameter in the oauth authorisation code flow. (auth code grant, auth code + features = flow)